packages/arb-token-bridge-ui/src/util/TokenUtils.ts

```diff
@@ -288,7 +324,19 @@ export async function l1TokenIsDisabled({
   l1Provider: Provider
   l2Provider: Provider
 }): Promise<boolean> {
-  const erc20Bridger = await Erc20Bridger.fromProvider(l2Provider)
+  const erc20Bridger = await getBridger({
+    sourceChainId: await getChainIdFromProvider(l1Provider),
+    destinationChainId: await getChainIdFromProvider(l2Provider)
+  })
+
+  if (
+    erc20Bridger instanceof EthL1L3Bridger ||
+    erc20Bridger instanceof EthBridger
+  ) {
+    // fail-safe to ensure `l1TokenIsDisabled` is called on the correct bridger-types
+    return false
+  }
```

Comment:
What about returning false in `EthL1L3Bridger/EthBridger` instead? So from the bridge UI point of view, we don't need to check for instance.

Reply:
Created FS-557 after our followup discussion with the team.

Reply:
i tried out sth but not sure if this is the solution we will want to go with:

<img width="1528" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/13184582/2a8ad52e-6628-4db5-b2fc-70a58788a47d">

with conditional types we can remove this snippet but it'll require casting within `getBridger`